src/Image.cpp

```diff
@@ -15,19 +15,29 @@ void ImageBase::prepare_for_direct_pixel_access() {
     buffer.set_host_dirty(true);
 
     if (buffer.defined()) {
+        // Cache some of the things we need to do to get efficient
+        // (low-dimensional) pixel access.
         origin = buffer.host_ptr();
-        stride_0 = buffer.stride(0);
-        stride_1 = buffer.stride(1);
-        stride_2 = buffer.stride(2);
-        stride_3 = buffer.stride(3);
+        ssize_t offset = 0;
+        for (int i = 0; i < buffer.dimensions(); i++) {
+            offset += buffer.dim(i).min() * buffer.dim(i).stride();
+        }
+        if (buffer.dimensions() > 0) {
+            stride_0 = buffer.dim(0).stride();
```

Comment:
A few bits of code seem like they would be cleaner if stride_n were an array instead.